### PR TITLE
[QF-4240] fix theme flicker

### PR DESCRIPTION
**Pull request by afifvdin**
## Summary

Fix theme flicker/flash that occurs when the locale (language) changes. The issue happened because Redux-persist's `REHYDRATE` action is asynchronous, causing the theme to briefly reset to default before the persisted state is restored.

The fix reads the persisted theme directly from `localStorage` synchronously before the Redux store is created, ensuring the user's theme preference is preserved immediately when the store is recreated during language switching.

Closes: [QF-4240](https://quranfoundation.atlassian.net/browse/QF-4240)

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] ♻️ Refactoring (no functional changes)

## Scope Confirmation

- [x] This PR addresses **one** feature/fix only
- [ ] If multiple changes were needed, they are split into separate PRs

## Rollback Safety

- [x] Can be safely reverted without data issues or migrations
- [ ] Rollback requires special steps (describe below):

## Test Plan

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing performed

**Testing steps:**

1. Set the theme to dark mode (or any non-default theme)
2. Change the language/locale from the settings
3. Observe that the theme remains consistent without any flash/flicker to the default theme
4. Refresh the page and verify theme persists correctly

### Edge Cases Verified

- [ ] ⏳ Loading state handled
- [ ] ❌ Error state handled (try-catch with console error logging)
- [ ] 📭 Empty state handled
- [ ] 👤 Logged-in vs guest behavior (if applicable)

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [ ] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [x] Build succeeds (`yarn build`)
- [x] Edge cases and error scenarios are handled

### Documentation

- [x] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [x] Inline comments added for complex logic

### Localization (if UI changes)

- N/A (no UI text changes)

### Accessibility (if UI changes)

- N/A (no UI changes)

## AI Assistance Disclosure

- [x] AI tools were NOT used for this PR
- [ ] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-4240]: https://quranfoundation.atlassian.net/browse/QF-4240?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/redux/utils/getPersistedTheme.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// changes from 'root', this constant must be updated accordingly.
+const PERSIST_KEY = 'persist:root';
+
+/**
+ * Get persisted theme from localStorage to prevent theme flash when locale changes.
+ * This ensures the user's current theme is preserved when the store is recreated.
+ *
+ * @returns {RootState[SliceName.THEME] | null} The persisted theme state or null if not found
+ */
+const getPersistedTheme = (): RootState[SliceName.THEME] | null => {
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/redux/utils/getPersistedTheme.ts:15]: done

File: src/redux/utils/getPersistedTheme.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      return null;
+    }
+
+    const parsed = JSON.parse(persistedState);
+    const themeState = parsed[SliceName.THEME];
+    if (!themeState) {
+      return null;
+    }
+
+    return JSON.parse(themeState);
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/redux/utils/getPersistedTheme.ts:35]: explained in the comment
